assets/src/checkout/pages/BamboraPayPage.tsx

```diff
@@ -102,20 +161,58 @@ const BamboraPayPage: React.FC<Props> = ({ path, gatewayAccount, payment }) => {
     expiry: generateId("card-expiry-")
   });
 
+  const [fieldComplete, setFieldComplete] = React.useState<null | {
+    field: Field;
+  }>(null);
+  const [fieldError, setFieldError] = React.useState<null | {
+    field: Field;
+  }>(null);
   const [cardErrors, setCardErrors] = React.useState<Record<Field, string>>({
     "card-number": "",
     cvv: "",
     expiry: ""
   });
+  const [cardBrand, setCardBrand] = React.useState<null | Brand>(null);
+  const cardTypesQuery = useCardTypesQuery({ errorPolicy: "all" });
```

Comment:
include cardtypes fragment in `usePaymentQuery` in the router above this and pass card types down. no need to do another network request here and we're mixing contexts by importing from admin